### ISSUE TITLE
mbox_type sticks to a filename for the duration of a session

### ISSUE THREAD

**Issue by nabijaczleweli** [neomutt/neomutt#2330]
## Expected Behaviour

The mailbox is created with the *current* `mbox_type`

## Actual Behaviour

If it was created previously with a different `mbox_type`, it'll be created with that one instead of a new one

## Steps to Reproduce

1. `:set mbox_type="MMDF"`
2. `<copy-message>`
3. Copy to mailbox: `~/uwu/does_not_exist`
4. Create /home/nabijaczleweli/uwu/does_not_exist? ([yes]/no): `y`
5. Copying to /home/nabijaczleweli/uwu/does_not_exist...
6. `^Z`; $ `stat ~/uwu/does_not_exist`
```
  File: /home/nabijaczleweli/uwu/does_not_exist
  Size: 77976           Blocks: 160        IO Block: 4096   **regular file**
Device: 36h/54d Inode: 33128664    Links: 1
```
7. $ `rm -rf ~/uwu/does_not_exist`; $ `fg`
8. `:set mbox_type="MH"`
9. `<copy-message>`
10. Copy to mailbox: `~/uwu/does_not_exist`
11. Create /home/nabijaczleweli/uwu/does_not_exist? ([yes]/no): `y`
12. Copying to /home/nabijaczleweli/uwu/does_not_exist...
13. `^Z`; $ `stat ~/uwu/does_not_exist`
```
  File: /home/nabijaczleweli/uwu/does_not_exist
  Size: 77976           Blocks: 160        IO Block: 4096   **regular file**
Device: 36h/54d Inode: 33128664    Links: 1
```

If you do it the other way around (i.e. first MH, then MMDF), the `does_not_exist` file will be a directory both times instead.

Writing to a previously-unused mailbox always uses the current `mbox_type`.

## How often does this happen?

- Always

## When did it start to happen?

dunno

## NeoMutt Version

20200501 + #2314 + #2316 + #2323 + #2328 + #2329

```
nabijaczleweli@tarta:~/uwu/neomutt-test$ ./pacc/usr/bin/neomutt -v
NeoMutt 20200501
Copyright (C) 1996-2016 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 4.19.0-8-amd64 (x86_64)
ncurses: ncurses 6.1.20181013 (compiled with 6.1.20181013)
libidn: 1.33 (compiled with 1.33)
GPGME: 1.12.0
GnuTLS: 3.6.7
libnotmuch: 5.2.0
storage: tokyocabinet
compression: zlib, zstd

Configure options: --build=x86_64-linux-gnu --prefix=/usr {--includedir=${prefix}/include} {--mandir=${prefix}/share/man} {--infodir=${prefix}/share/info} --sysconfdir=/etc --localstatedir=/var --disable-silent-rules {--libdir=${prefix}/lib/x86_64-linux-gnu} {--libexecdir=${prefix}/lib/x86_64-linux-gnu} --disable-maintainer-mode --disable-dependency-tracking --mandir=/usr/share/man --libexecdir=/usr/libexec --with-mailpath=/var/mail --gpgme --lua --notmuch --with-ui --gnutls --gss --idn --mixmaster --sasl --tokyocabinet --zlib --zstd --testing

Compilation CFLAGS: -g -O2 -fdebug-prefix-map=/home/nabijaczleweli/uwu/neomutt-test/neomutt.deb=. -fstack-protector-strong -Wformat -Werror=format-security -std=c99 -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/usr/include -I/usr/include/lua5.3 -DNCURSES_WIDECHAR -isystem /usr/include/mit-krb5

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash

Compile options:
  -autocrypt +bkgdset +color +curs_set +fcntl -flock -fmemopen +futimens
  +getaddrinfo +gnutls +gpgme +gss +hcache -homespool +idn +inotify
  -locales_hack +lua -lz4 +meta +mixmaster +nls +notmuch -openssl +pgp +regex
  +sasl +smime -sqlite +start_color +sun_attachment +typeahead +zlib +zstd
MAILPATH="/var/mail"
MIXMASTER="mixmaster"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

**Comment by flatcap** [comment]
Hmm... that's certainly an interesting bug.  It exists in upstream mutt, too.

When you first save the email, NeoMutt creates a `Mailbox` associated with the path.
This `Mailbox` has type `$mbox_type`.

NeoMutt isn't aware that you've deleted the file.
When you try to save to the same path, NeoMutt finds a `Mailbox` that matches the path and uses it.
`$mbox_type` doesn't get consulted.

Deep thought is required :-)

**Comment by nabijaczleweli** [comment]
Fixed as of 3c5d6951981b4014b8a8cb9c272a2185212cb129
